src/items/generics.md

```diff
@@ -0,0 +1,107 @@
+# Type and Lifetime Parameters
+
+> **<sup>Syntax</sup>**  
+> _Generics_ :  
+> &nbsp;&nbsp; `<` _GenericParams_ <sup>?</sup> `>`  
+>  
+> _GenericParams_ :  
+> &nbsp;&nbsp; &nbsp;&nbsp; _LifetimeParams_  
+> &nbsp;&nbsp; | ( _LifetimeParams_ `,` ) <sup>?</sup>
+> _TypeParams_ `,` <sup>?</sup>  
+>  
+> _LifetimeParams_ :  
+> &nbsp;&nbsp; _LifetimeParam_ (`,` _LifetimeParam_)<sup>\*</sup>  
+>  
+> _LifetimeParam_ :  
+> &nbsp;&nbsp; [LIFETIME_OR_LABEL] [_LifetimeBounds_]<sup>?</sup>  
+>  
+> _Lifetime_ :  
+> &nbsp;&nbsp; &nbsp;&nbsp; [LIFETIME_OR_LABEL]  
+> &nbsp;&nbsp; | `'static`  
+>  
+> _TypeParams_:  
+> &nbsp;&nbsp; _TypeParam_ (`,` _TypeParam_)<sup>\*</sup>  
+>  
+> _TypeParam_ :  
+> &nbsp;&nbsp; [IDENTIFIER] [_TypeParamBounds_]<sup>?</sup> ( `=` [_Type_] )<sup>?</sup>  
+
+Functions, type aliases, structs, enumerations, unions, traits and
+implementations may be *parameterized* by types and lifetimes. These parameters
+are listed in angle brackets (`<...>`), after the name of the item (except for
+implementations, where they come directly after `impl`) and before its
+definition. Lifetime parameters must be declared before type parameters.
+Some examples of items with type and lifetime parameters:
+
+```rust
+fn foo<'a, T>() {}
+trait A<U> {}
+struct Ref<'a, T> where T: 'a { r: &'a T }
+```
+
+Certain built in types: [references], [raw pointers], [arrays],
+[slices][arrays], [tuples] and [function pointers], have lifetime or type
```

Comment:
Nit: built-in*

Nit: trailing comma after "function pointers"

Less nitty: Is the prefix of saying they are built-in types necessary here? The text would flow more smoothly if you just started at "References", IMO.